File: frontend/src/app/core/models/tickets/get-ticket-dto.ts
import { TicketTargetType } from "../enums/target-type";
import { GetTicketStatusDto } from "./get-ticket-status-dto";

export interface GetTicketDto {
    ticketId: string;
    submitterId: string;
    targetId: string;
    targetType: TicketTargetType;
    reason: string;
    status: GetTicketStatusDto;
    response: string;
  }

File: backend/YouTubeV2.Application/DTO/TicketDTOS/GetTicketDto.cs
using YouTubeV2.Application.Enums;
using YouTubeV2.Application.Model;

namespace YouTubeV2.Application.DTO.TicketDTOS
{
    public record GetTicketDto(Guid TicketId, Guid SubmitterId, Guid TargetId, TicketTargetType TargetType, string Reason, GetTicketStatusDto Status, string Response)
    {
        public GetTicketDto(Ticket ticket, Guid submitterId) : this(ticket.Id, submitterId, ticket.TargetId, ticket.TargetType, ticket.Reason, new GetTicketStatusDto(ticket.Status), ticket.Response) { }
    }
}

